pkg/tag/api/api.go

```diff
@@ -121,8 +121,42 @@ func (s *TagService) CreateTag(
 		)
 		return nil, s.reportInternalServerError(ctx, err, req.EnvironmentId, localizer)
 	}
-	// Save in the DB
-	if err := s.tagStorage.UpsertTag(ctx, tag); err != nil {
+
+	var event *eventproto.Event
+	var actualTag *domain.Tag
+	err = s.mysqlClient.RunInTransactionV2(ctx, func(ctxWithTx context.Context, _ mysql.Transaction) error {
```

Comment:
The race condition happened because the upsert process wasn't done inside a transaction.